content/docs/user-guide/build/generated-source.md

```diff
@@ -0,0 +1,145 @@
+)
+```
+
+Data input files and templates are included in [Code\Framework\AzNetworking\AzNetworking\aznetworking_files.cmake](https://github.com/o3de/o3de/blob/dd0978c59f1d01b39e006e6c3ba3baf6060136cf/Code/Framework/AzNetworking/AzNetworking/aznetworking_files.cmake#L12-L17). 
+```cmake
+set(FILES
+    # ...
+    AutoGen/AutoPacketDispatcher_Header.jinja
+    AutoGen/AutoPacketDispatcher_Inline.jinja
+    AutoGen/AutoPackets_Header.jinja
+    AutoGen/AutoPackets_Inline.jinja
+    AutoGen/AutoPackets_Source.jinja
+    AutoGen/CorePackets.AutoPackets.xml
+    # ...
+)
+```
+
+
+## Autogen rules
+
+An *autogen rule* instructs CMake to use AzAutoGen to generate files. 
+You define autogen rules by passing in a set of `AUTOGEN_RULES` to the `ly_add_target(...)` command that's inside your project's `CMakeLists.txt` file.
+
+Each element of the set passed to `AUTOGEN_RULES` should follow the `<input>,<template>,<output>` format. Here, `<input>` is the name of the XML or JSON data input file, `<template>` is the name of the Jinja template file, and `<output>` is the name that you want the generated file to have. 
+
+{{< important >}}
+Data input files and Jinja template files must be listed in your project's set of files, which is located in a `*_files.cmake` file, inside a `set(FILES...)` command. For organization, we recommend creating a specific `*_autogen_files.cmake` file and placing Jinja template files there. 
+{{< /important >}}
+
+### Data input files
+
+A data input filename can refer to one file explicitly or many files that match a pattern.  If you list a filename explicitly, then it must have a proper path that's absolute or relative to the `CMakeLists.txt` file. If a filename contains a pattern, then the expansion finds all data input files that match the filename.
+
+<!-- TODO: Verify - can explicit filenames be absolute or relative to the CMakeLists.txt file?-->
+
+The input filename supports the following matching operators:
+  * `*` - Sequence of characters any length.
+  * `?` - Single-character sequence.
+  * `[<sequence>]` - Matches any characters in `<sequence>`.
+  * `[!<sequence>]` - Matches any characters _not_ in `<sequence>`.
+
+For information about authoring data input files, refer to [Authoring Jinja templates and data inputs](/docs/user-guide/programming/autogen/).
+
+### Jinja Templates
+
+A template filename must be an explicit path to a single `.jinja` file. The path can be absolute or relative to the `CMakeLists.txt` file. 
+
+For information about authoring Jinja templates, refer to [Authoring Jinja templates and data inputs](/docs/user-guide/programming/autogen/).
+
+
+### Output files
+
+When AzAutoGen feeds data input files into the Jinja2 templates, it generates output files. You must specify the names of the output files you want to generate in the autogen rules. You can use the following *special values* when defining the output filenames:
+
+* `$path`: The path to the final output destination.
```

Comment:
We can reiterate that the `$path` corresponds to the `${CMAKE_CURRENT_BINARY_DIR}/Azcg/Generated` directory from line 25 of this doc.